src/object.d

```diff
@@ -3722,6 +3722,91 @@ template _arrayOp(Args...)
     alias _arrayOp = arrayOp!Args;
 }
 
+// switch template - replaces code in rt/switch_.d
+int __switch(T)(T[] condition, T[][] caseLabels...)
+{
+    for (uint i; i < caseLabels.length; i++)
```

Comment:
Similar to _d_switch_string this should take advantage of the string array being sorted, i.e. copy the binary search. If the array size could be made a part of the template arguments, a switch with few cases could be special cased to linear search.

Reply:
Indeed.
Also using a variation of Trie comes to mind - at least a level or two of switching on the first (second) character may help a bit.

Other tricks involve first switching on condition.length.